docs/spec/concepts.rst

```diff
@@ -3,97 +3,216 @@
+checker should not emit any errors that depend on assuming a particular static
+type; a static checker should instead assume that the runtime is responsible
+for checking the type of operations on this term, as in a dynamically-typed
+language.
+
+The gradual guarantee
+~~~~~~~~~~~~~~~~~~~~~
+
+:ref:`Any` allows gradually adding static types to a dynamically-typed program.
+For a fully dynamically-typed program, a static checker has the type :ref:`Any`
+for all terms, and should emit no errors. Adding more static type annotations
+to the program (making the program more statically typed) may result in static
+type errors, if the program is not correct or if the static type annotations
+aren't able to fully represent the runtime types. Removing static type
+annotations (making the program more dynamic) should not result in additional
+static type errors. This is often referred to as the **gradual guarantee**.
+
+In Python's type system, we don't take the gradual guarantee as a strict
+requirement, but it's a useful guideline.
+
+The subtype relation
+--------------------
+
+A static type ``B`` is a **subtype** of another static type ``A`` if and only
+if the set of values represented by ``B`` is a subset of the set of values
+represented by ``A``. Because the subset relation on sets is transitive and
+reflexive, the subtype relation is also transitive (if ``C`` is a subtype of
+``B`` and ``B`` is a subtype of ``A``, then ``C`` is a subtype of ``A``) and
+reflexive (``A`` is always a subtype of ``A``).
+
+The **supertype** relation is the inverse of subtype: ``A`` is a supertype of
+``B`` if and only if ``B`` is a subtype of ``A``; or equivalently, if and only
+if the set of values represented by ``A`` is a superset of the values
+represented by ``B``. The supertype relation is also transitive and reflexive.
+
+We also define an **equivalence** relation on static types: the types ``A`` and
+``B`` are equivalent (or "the same type") if and only if ``A`` is a subtype
+of ``B`` and ``B`` is a subtype of ``A``. This means that the set of values
+represented by ``A`` is both a superset and a subset of the values represented
+by ``B``, meaning ``A`` and ``B`` must represent the same set of values.
+
+We may describe a type ``B`` as "narrower" than a type ``A`` (or as a "strict
+subtype" of ``A``) if ``B`` is a subtype of ``A`` and ``B`` is not equivalent
+to ``A``.
+
+The consistency relation
+------------------------
+
+Since :ref:`Any` represents an unknown static type, it does not represent any
+known single set of values, and thus it is not in the domain of the subtype,
+supertype, or equivalence relations on static types described above. (We do
+find it convenient to say, however, that ``Any`` is both subtype and supertype
+of -- that is, equivalent to -- only itself. This can allow us to simplify
+redundant multiple occurrences of ``Any`` out of more complex types.)
+
+To relate gradual types more generally, we define a **materialization**
```

Comment:
I also think it's a useful and intuitive concept! It was first introduced in
the literature (as far as I'm aware) in the 2019 paper "Gradual Typing: A New
Perspective." (https://www.irif.fr/~gc/papers/popl19.pdf)

I'm not sure how I can introduce it earlier and use it to define gradual types,
though. Since materialization is the process of replacing Any in a type with a
more static type, that seems to depend on first understanding that some types
can contain Any as a sub-part. If you're interested in this idea, maybe you can
suggest in a bit more detail how this presentation would look?

Reply:
I think your updated wording in the earlier section eliminates the need to introduce this concept earlier, so I withdraw this suggestion.